PWGJE/Tasks/nucleiWithFastJet.cxx

```diff
@@ -0,0 +1,1500 @@
+        if (jetFlag) {
+          jetHist.fill(HIST("mcpJet/helium/pt/PtHelium_jet"), mcParticle.pt());
+        } else {
+          jetHist.fill(HIST("mcpJet/helium/pt/PtHelium_outJet"), mcParticle.pt());
+        }
+      }
+
+      if (mcParticle.pdgCode() == -PDGHelium) {
+        jetHist.fill(HIST("mcpJet/antiHelium/pt/PtantiHelium"), mcParticle.pt());
+        if (jetFlag) {
+          jetHist.fill(HIST("mcpJet/antiHelium/pt/PtantiHelium_jet"), mcParticle.pt());
+        } else {
+          jetHist.fill(HIST("mcpJet/antiHelium/pt/PtantiHelium_outJet"), mcParticle.pt());
+        }
+      }
+
+      if (mcParticle.pdgCode() == PDGTriton) {
+        jetHist.fill(HIST("mcpJet/triton/pt/PtTriton"), mcParticle.pt());
+        if (jetFlag) {
+          jetHist.fill(HIST("mcpJet/triton/pt/PtTriton_jet"), mcParticle.pt());
+        } else {
+          jetHist.fill(HIST("mcpJet/triton/pt/PtTriton_outJet"), mcParticle.pt());
+        }
+      }
+
+      if (mcParticle.pdgCode() == -PDGTriton) {
+        jetHist.fill(HIST("mcpJet/antiTriton/pt/PtantiTriton"), mcParticle.pt());
+        if (jetFlag) {
+          jetHist.fill(HIST("mcpJet/antiTriton/pt/PtantiTriton_jet"), mcParticle.pt());
+        } else {
+          jetHist.fill(HIST("mcpJet/antiTriton/pt/PtantiTriton_outJet"), mcParticle.pt());
+        }
+      }
+    } // track
+  }   // process mc
+
+  void processMCRec(o2::aod::JCollision const& collisionJet, soa::Join<aod::JTracks, aod::JTrackPIs, aod::McTrackLabels> const& tracks,
+                    soa::Filtered<aod::ChargedMCDetectorLevelJets> const& mcdjets, aod::McParticles const&, TrackCandidatesMC const&)
+  {
+    jetHist.fill(HIST("mcdJet/eventStat"), 0.5);
+    // JEhistos.fill(HIST("nEvents_MCRec"), 0.5);
+
+    if (!jetderiveddatautilities::selectCollision(collisionJet, jetderiveddatautilities::JCollisionSel::sel8))
+      return;
+    // bool jetFlag = kFALSE;
+
+    jetHist.fill(HIST("mcdJet/eventStat"), 1.5);
+
+    if (fabs(collisionJet.posZ()) > 10)
+      return;
+
+    jetHist.fill(HIST("mcdJet/eventStat"), 2.5);
+    bool INELgt0 = false;
+    for (const auto& track : tracks) {
+      if (fabs(track.eta()) < cfgtrkMaxEta) {
+        INELgt0 = true;
+        break;
+      }
+    }
+    if (!INELgt0)
+      return;
+    jetHist.fill(HIST("mcdJet/eventStat"), 3.5);
+    int nJets = 0;
+    for (auto& mcdjet : mcdjets) {
+      jetHist.fill(HIST("mcdJet/hJetPt"), mcdjet.pt());
+      jetHist.fill(HIST("mcdJet/hJetEta"), mcdjet.eta());
+      jetHist.fill(HIST("mcdJet/hJetPhi"), mcdjet.phi());
+      nJets++;
+    }
+
+    jetHist.fill(HIST("mcdJet/vertexZ"), collisionJet.posZ());
+    jetHist.fill(HIST("mcdJet/nJetsPerEvent"), nJets);
+
+    for (const auto& track : tracks) {
+      auto fullTrack = track.track_as<TrackCandidatesMC>();
+
+      if (!isTrackSelected(fullTrack))
+        continue;
+      if (fabs(fullTrack.eta()) > cfgtrkMaxEta)
+        continue;
+      if (!track.has_mcParticle())
```

Comment:
you can also get these parameters from JParticles. Then you wouldnt need to subscribe to the aod::McParticles table

Reply:
It it now done.